src/Diagrams/Attributes.hs

```diff
@@ -338,3 +410,39 @@ lineMiterLimit = applyAttr . LineMiterLimit . Last
 lineMiterLimitA :: HasStyle a => LineMiterLimit -> a -> a
 lineMiterLimitA = applyAttr
 
+_lineMiterLimit :: Lens' (Style v n) Double
+_lineMiterLimit = atAttr . mapping _LineMiterLimit . non 10
+
+------------------------------------------------------------------------
+-- Recommend optics
+------------------------------------------------------------------------
+
+-- | Prism onto a 'Recommend'.
+_Recommend :: Prism' (Recommend a) a
+_Recommend = prism' Recommend $ \case (Recommend a) -> Just a; _ -> Nothing
+
+-- | Prism onto a 'Commit'.
+_Commit :: Prism' (Recommend a) a
+_Commit = prism' Commit $ \case (Commit a) -> Just a; _ -> Nothing
+
+-- | Lens onto the value inside either a 'Recommend' or 'Commit'. Unlike
+--   'committed', this is a valid lens.
+_recommend :: Lens (Recommend a) (Recommend b) a b
+_recommend f (Recommend a) = Recommend <$> f a
+_recommend f (Commit a)    = Commit <$> f a
+
+-- | Lens onto weather something is committed or not.
```

Comment:
If only we had a lens into weather...

On Fri, Feb 27, 2015 at 10:19 AM, Daniel Bergey <notifications@github.com>
wrote:

> In src/Diagrams/Attributes.hs
> <https://github.com/diagrams/diagrams-lib/pull/236#discussion_r25512596>:
>
> > +
> > +-- | Prism onto a 'Recommend'.
> > +_Recommend :: Prism' (Recommend a) a
> > +_Recommend = prism' Recommend $ \case (Recommend a) -> Just a; _ -> Nothing
> > +
> > +-- | Prism onto a 'Commit'.
> > +_Commit :: Prism' (Recommend a) a
> > +_Commit = prism' Commit $ \case (Commit a) -> Just a; _ -> Nothing
> > +
> > +-- | Lens onto the value inside either a 'Recommend' or 'Commit'. Unlike
> > +--   'committed', this is a valid lens.
> > +_recommend :: Lens (Recommend a) (Recommend b) a b
> > +_recommend f (Recommend a) = Recommend <$> f a
> > +_recommend f (Commit a)    = Commit <$> f a
> > +
> > +-- | Lens onto weather something is committed or not.
>
> whether, not weather
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/diagrams/diagrams-lib/pull/236/files#r25512596>.
>